### ISSUE TITLE
add heart-beat support to avoid time out disconnect

### ISSUE THREAD

**Issue by 1037390459** [NaikSoftware/StompProtocolAndroid#88]

**Comment by forresthopkinsa** [comment]
Hey, make an issue on the [downstream](https://github.com/forresthopkinsa/StompProtocolAndroid) explaining what you want with as much detail as you can manage, and we'll start looking into our options for making this work well.

**Comment by luisalves00** [comment]
I'm adding heart-beat support. Will you accept a pull request and do a new release?

**Comment by NaikSoftware** [comment]
Yes of course

**Comment by jdupont22** [comment]
Hi,

I've found a trick to avoid server timeout disconnect on missing pong reply on ping frame from the server. Only work with "Stomp.ConnectionProvider.OKHTTP".

I've modified the class "OkHttpConnectionProvider" on line 67, when the heartbeat from serveur is received, like this:

`
@Override
public void onMessage(WebSocket webSocket, String text) {
    if (text.equals("\n")) {
        Log.d(TAG, "RECEIVED HEARTBEAT");
        rawSend("\n");
    } else
        emitMessage(text);
}
`
On each heartbeat from the server, i respond with a raw frame "\n".

Work with RabbitMq and WebStomp plugin.
